### PR TITLE
=doc #18968 Document auth options for Java DSL

### PR DESCRIPTION
**Pull request by johanandren**
Refs #18968

### CODE REVIEW

**Comment by ktoso** [comment]
PLS BUILD

**Comment by ktoso** [comment]
lol, just noticed this was PRed 2 minutes ago; was here to PLS BUILD before jenkins had the chace to ;-)

File: akka-docs-dev/rst/java/http/routing-dsl/request-vals/oauth2-authenticator.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+An abstract class to implement Oauth 2 bearer token authentication
+
+Description
+-----------
+Allows to protect one of more routes with authentication in the form of a OAuth2 Bearer Token. For more information
+about OAuth 2 Bearer Token see `RFC6750`_.
+
+.. _RFC6750: https://tools.ietf.org/html/rfc6750
+
+To use it you subclass ``OAutht2Authenticator`` and implement the ``authenticate`` method
```
Comment (ktoso) [inline akka-docs-dev/rst/java/http/routing-dsl/request-vals/oauth2-authenticator.rst]: hint, we sometimes use 

```
:class:`OAutht2Authenticator`
```

(currently does nothing, but we're hoping to get tooling around it eventually)

**Comment by ktoso** [comment]
LGTM, thanks a lot!

File: akka-docs-dev/rst/java/http/routing-dsl/request-vals/oauth2-authenticator.rst (hunk 1 above)
Comment (johanandren) [inline akka-docs-dev/rst/java/http/routing-dsl/request-vals/oauth2-authenticator.rst]: argh. the spelling! Oautht

Comment (ktoso) [inline akka-docs-dev/rst/java/http/routing-dsl/request-vals/oauth2-authenticator.rst]: :< Didn't notice, sorry.

Comment (ktoso) [inline akka-docs-dev/rst/java/http/routing-dsl/request-vals/oauth2-authenticator.rst]: I'll fix that.

**Pull request merged**